main.tf

```diff
@@ -445,6 +445,20 @@ resource "aws_emr_cluster" "default" {
     }
   }
 
+  dynamic "step" {
+    for_each = var.steps
+    content {
+      name                                = step.value.name
+      action_on_failure                   = step.value.action_on_failure
+      hadoop_jar_step {
+        jar                               = step.value.hadoop_jar_step["jar"]
+        main_class                        = step.value.hadoop_jar_step["main_class"]
```

Comment:
Can you add something like this `lookup(step.value.hadoop_jar_step, "main_class", null)` for non-required params?

Reply:
@SweetOps changed it to lookup + changed the issue title